elk/extraction/prompt_loading.py

```diff
@@ -165,29 +64,47 @@ def load_prompts(
     if rank == 0:
         print(f"Using {num_variants} variants of each prompt")
 
-    label_column = label_column or infer_label_column(ds.features)
-    num_classes = num_classes or infer_num_classes(ds.features[label_column])
-    rng = Random(seed)
+    label_column = prompter.label_column or infer_label_column(ds.features)
 
+    label_feature = ds.features[label_column]
+    if isinstance(label_feature, ClassLabel):
+        label_choices = [label_feature.str2int(label) for label in label_feature.names]
+    elif isinstance(label_feature, Value) and label_feature.dtype == "bool":
+        label_choices = [False, True]
+    else:
+        # Which classes are actually present in this split of the dataset?
+        # This is shockingly fast since it uses an optimized Apache Arrow primitive.
+        label_choices = sorted(ds.unique(label_column))
```

Comment:
Perhaps we should warn that we're assuming all class labels are represented?

Reply:
Yes

Reply:
fixed